src/test/java/org/sagebionetworks/bridge/services/FileServiceTest.java

```diff
@@ -218,5 +257,126 @@ public void deleteAllStudyFiles() {
+        
+        verify(mockFileRevisionDao).getFileRevisions(GUID, 100, 25);
+    }
+    
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getFileRevisionsPageSizeTooSmall() {
+        FileMetadata metadata = new FileMetadata();
+        doReturn(metadata).when(service).getFile(TEST_STUDY, GUID);
+        
+        service.getFileRevisions(TEST_STUDY,GUID, 0, 1);
+    }
+    
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getFileRevisionsSizeTooLarge() {
+        FileMetadata metadata = new FileMetadata();
+        doReturn(metadata).when(service).getFile(TEST_STUDY, GUID);
+        
+        service.getFileRevisions(TEST_STUDY, GUID, 0, 1000);
+    }
+    
+    @Test
+    public void createFileRevision() throws Exception {
+        FileMetadata metadata = new FileMetadata();
+        doReturn(metadata).when(service).getFile(TEST_STUDY, GUID);
+        
+        URL url = new URL("https://" + UPLOAD_BUCKET);
+        when(mockS3Client.generatePresignedUrl(any())).thenReturn(url);
+        
+        FileRevision revision = new FileRevision();
+        revision.setName("name.pdf");
+        revision.setFileGuid(GUID);
+        revision.setMimeType("application/pdf");
+        
+        FileRevision returned = service.createFileRevision(TEST_STUDY, revision);
+        assertEquals(returned.getCreatedOn(), TIMESTAMP);
+        assertEquals(returned.getStatus(), PENDING);
+        assertEquals(returned.getUploadURL(), "https://" + UPLOAD_BUCKET);
+        assertEquals(returned.getDownloadURL(), "https://docs.sagebridge.org/oneGuid.1422319112486");
+        
+        verify(mockS3Client).generatePresignedUrl(requestCaptor.capture());
+        GeneratePresignedUrlRequest request = requestCaptor.getValue();
+        assertEquals(request.getExpiration().getTime(), TIMESTAMP.plusMinutes(EXPIRATION_IN_MINUTES).getMillis());
+        assertEquals(request.getBucketName(), UPLOAD_BUCKET);
+        assertEquals(request.getKey(), GUID + "." + TIMESTAMP.getMillis());
+        assertEquals(request.getContentType(), "application/pdf");
+        assertEquals(request.getResponseHeaders().getContentDisposition(), "attachment; filename=\"name.pdf\"");
+        
+        verify(mockFileRevisionDao).createFileRevision(revisionCaptor.capture());
+        assertSame(revisionCaptor.getValue(), returned);
+    }
+    
+    @Test(expectedExceptions = EntityNotFoundException.class, 
+            expectedExceptionsMessageRegExp=".*FileMetadata not found.*")
+    public void createFileRevisionNotFound() throws Exception {
+        FileRevision revision = new FileRevision();
+        revision.setFileGuid(GUID);
+        
+        service.createFileRevision(TEST_STUDY, revision);
+    }
+        
+    @Test
+    public void finishFileRevision() {
+        FileRevision existing = new FileRevision();
+        existing.setFileGuid(GUID);        
+        existing.setCreatedOn(TIMESTAMP);
```

Comment:
Should we set a dummy upload URL so we can verify that it's actually being blanked out?

Reply:
Added